tests/test_blocks.py

```diff
@@ -69,6 +71,106 @@ def test_mpl_kwargs(self):
         assert actual._mpl_kwargs == expected
 
 
+class TestLineBlock:
+    def test_2d_inputs(self):
+        x = np.linspace(0, 1, 10)
+        t = np.linspace(0, 1, 5)
+        x_grid, t_grid = np.meshgrid(x, t)
+        y_data = np.sin(2 * np.pi * (x_grid + t_grid))
+
+        line_block = amp.blocks.Line(x_grid, y_data)
+
+        assert isinstance(line_block, amp.blocks.Line)
+        npt.assert_equal(line_block.x, x_grid)
+        npt.assert_equal(line_block.y, y_data)
+        assert len(line_block) == len(t)
+
+        assert isinstance(line_block.line, mpl.lines.Line2D)
+        xdata, ydata = line_block.line.get_data()
+        npt.assert_equal(xdata, x)
+        npt.assert_equal(ydata, y_data[0, :])
+
+    def test_update(self):
+        x = np.linspace(0, 1, 10)
+        t = np.linspace(0, 1, 5)
+        x_grid, t_grid = np.meshgrid(x, t)
+        y_data = np.sin(2 * np.pi * (x_grid + t_grid))
+
+        line_block = amp.blocks.Line(x_grid, y_data)
+        line_block._update(frame=1)
+
+        npt.assert_equal(line_block.line.get_xdata(), x)
+        npt.assert_equal(line_block.line.get_ydata(), y_data[1, :])
+
+    def test_constant_x(self):
+        x = np.linspace(0, 1, 10)
+        t = np.linspace(0, 1, 5)
+        x_grid, t_grid = np.meshgrid(x, t)
+        y_data = np.sin(2 * np.pi * (x_grid + t_grid))
+
+        line_block = amp.blocks.Line(y_data)
+
+        expected_x = np.arange(10)
+        npt.assert_equal(line_block.line.get_xdata(), expected_x)
+
+    def test_no_x_input(self):
+        x = np.linspace(0, 1, 10)
+        t = np.linspace(0, 1, 5)
+        x_grid, t_grid = np.meshgrid(x, t)
+        y_data = np.sin(2 * np.pi * (x_grid + t_grid))
+
+        line_block = amp.blocks.Line(y_data)
+
+        expected_x = np.arange(10)
+        npt.assert_equal(line_block.line.get_xdata(), expected_x)
+
+    def test_list_input(self):
```

Comment:
This should specifically test for lists with arrays of different length inside.

If the y-data is a list of arrays of non-uniform length and no x-data is passed, we could either:
1. throw an error (force x-values to be passed) (after parsing ```y=np.asanyarray(...)```, we can check if it is of ```dtype=object``` or ```len(shape)!=2``` if not then is it a list of arrays of different lengths) or
2. create a list of x-data of arrays using arrange. 

I'd say to go for 1 since it is simpler.

Reply:
I agree, and would in fact say that 2 isn't possible because it's underspecified: e.g. If the second timestep has more y data values than the first, does that mean the user wants to plot a longer line or a line with more resolution in the second frame?